src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -712,13 +746,14 @@ impl Coordinator {
                                 timestamp_context,
                                 optimizer,
                                 global_lir_plan,
+                                peek_ctx,
                             })
                         }
                     }
                     // Internal optimizer errors are handled differently
                     // depending on the caller.
                     Err(err) => {
-                        let Some(explain_ctx) = explain_ctx else {
+                        let PeekContext::Explain(explain_ctx) = peek_ctx else {
```

Comment:
I would expect to see the CopyTo variant here as like a PeekStage::CopyTo. From my read, the stuff in peek_stage_finish is not needed by copy?

Reply:
`peek_stage_finish` is what finally calls the `implement...` method to schedule the dataflow. Since even copy to needs a dataflow, I forked the logic there instead of creating a different stage. It's also emitting optimizer notices, doing some `install_watch_set` calls which I thought might be needed for the copy to as well?

Reply:
Separating this out in a different finish stage for copy_to